src/utils/focusController.ts

```diff
@@ -15,94 +15,89 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { uuidv4 } from "@/types/helpers";
-import type { UUID } from "@/types/stringTypes";
-
-const FOCUS_CONTROLLER_UUID_SYMBOL = Symbol.for("focus-controller-uuid");
+import { isLoadedInIframe } from "@/utils/iframeUtils";
 
 declare global {
-  interface Window {
-    [FOCUS_CONTROLLER_UUID_SYMBOL]?: UUID;
-  }
+  // eslint-disable-next-line no-var, @shopify/prefer-module-scope-constants -- It must be a var for types to work
+  var PB_FOCUS_CONTROLLER: boolean;
 }
 
-class FocusController {
-  /**
-   * The last saved focused element
-   * @private
-   */
-  private focusedElement: HTMLElement | undefined;
-
-  /**
-   * Nonce to detect multiple imports of focusController
-   * @private
-   */
-  private readonly nonce = uuidv4();
+let focusedElement: HTMLElement | undefined;
 
-  constructor() {
-    // eslint-disable-next-line security/detect-object-injection -- symbol
-    if (window[FOCUS_CONTROLLER_UUID_SYMBOL]) {
-      console.warn(
-        // eslint-disable-next-line security/detect-object-injection -- symbol
-        `focusController(${this.nonce}): ${window[FOCUS_CONTROLLER_UUID_SYMBOL]} already added to window`,
-      );
-    } else {
-      // eslint-disable-next-line security/detect-object-injection -- symbol
-      window[FOCUS_CONTROLLER_UUID_SYMBOL] = this.nonce;
-    }
-  }
+if (globalThis.PB_FOCUS_CONTROLLER) {
+  console.error(
+    "focusController: there should only be one focusController instance per window",
+  );
```

Comment:
This should really be an error

```suggestion
  throw new Error(
    "focusController: there should only be one focusController instance per window"
  );
```